LSP-metals.sublime-commands

```diff
@@ -47,9 +47,10 @@
             "command_args": ["${file_uri}"]
         }
     },
-     {
+    {
         "caption": "LSP-metals: Show Build Target Info",
         "command": "lsp_metals_show_build_target_info",
         "args": {"command_name": "list-build-targets"}
-    }   
+    },
+    { "caption": "LSP-metals: Run Scalafix", "command": "lsp_metals_run_scalafix"}
```

Comment:
Again should be possible without a custom command by using `$position` as an argument.

Reply:
the command requires [TextDocumentPositionParams](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentPositionParams) while `$position` sends only the position

Reply:
Right, but you can construct the argument using `$document_id` and `$position`.

Reply:
These macros doesn't seems to work, this is what I used 
```
    {
        "caption": "LSP-metals: Run Scalafix",
        "command": "lsp_metals_execute",
        "args": {
            "command_name": "scalafix-run",
            "command_args": ["{'textDocument': ${document_id}, 'position': ${position}}"]
        }
    }
```
But in the logs I see empty values 
```
:: --> metals workspace/executeCommand(11): {'arguments': ["{'textDocument': , 'position': }"], 'command': 'scalafix-run', 'workDoneToken': 'wd11'}
```

Am I missing something ?

Reply:
Like this I think (not tested):

```js
    {
        "caption": "LSP-metals: Run Scalafix",
        "command": "lsp_metals_execute",
        "args": {
            "command_name": "scalafix-run",
            "command_args": [{"textDocument": "${document_id}", "position": "${position}"}]
        }
    }
```

Reply:
Hmm, looking at the source it seems like this will probably not work as it only processes top level array items.

OK, then you have to go with custom command then.
But it would be easy to add a template variable for `TextDocumentPositionParams` in LSP.